src/vsCodeConfig/launch.ts

```diff
@@ -28,7 +28,19 @@ export function updateLaunchVersion(folder: WorkspaceFolder, version: string): v
 }
 
 export function isDebugConfigEqual(c1: DebugConfiguration, c2: DebugConfiguration): boolean {
-    return c1.name === c2.name && c1.request === c2.request && c1.type === c2.type;
+    return c1.name === c2.name && c1.request === c2.request && isTypeEqual(c1.type, c2.type);
+}
+
+function isTypeEqual(type1: string, type2: string): boolean {
+    return type1 === type2 || (isNodeType(type1) && isNodeType(type2));
+}
+
+/**
+ * Special case node debug type because it can be either "node" or "node2"
+ * https://github.com/microsoft/vscode-azurefunctions/issues/1259
+ */
+function isNodeType(t: string): boolean {
+    return /^node2?$/i.test(t);
```

Comment:
Can't you just search for node instead of exclusively node2?  I'd expect if I had the "node" debug type for this to return true, but I don't think that'd be the case right now

Reply:
`?` is a quantifier meaning `2` is an optional character. This regexp matches both `node2` _and_ `node`